storage/redis_cluster.go

```diff
@@ -209,11 +220,15 @@ func (r RedisCluster) ensureConnection() {
 }
 
 // GetKey will retrieve a key from the database
-func (r RedisCluster) GetKey(keyName string) (string, error) {
+func (r *RedisCluster) GetKey(keyName string) (string, error) {
 	r.ensureConnection()
 	log.Debug("[STORE] Getting WAS: ", keyName)
 	log.Debug("[STORE] Getting: ", r.fixKey(keyName))
+	currentMu := r.getSingletonMU()
+	currentMu.RLock() //Lock rediscluster.RedisCluster.SingleRedisMode from writes
+	defer currentMu.RUnlock()
```

Comment:
as a general note - locking mutex, even read-locking with `RLock`, do some network I/O which might take ages and then unlock mutex - I feel like it is not right, this is not what mutexes for